### PR TITLE
[WIP] Streaming, again

### PR DESCRIPTION
**Pull request by charlag**
Cherry picked #892 which is beyond salvation by now.
Surprisingly it really works well, I think adding some refresh/gap thing and some disconnect/connect behavior could be enough to ship it. Maybe a toggle as well, although web doesn't have it.

### CODE REVIEW

**Comment by connyduck** [comment]
Ready for review?

**Comment by charlag** [comment]
I think nor yet, need to do something with timeline on resume

**Comment by charlag** [comment]
I've rebased it and added things to load top events on resume. Also I've tried to sync them a little. There's much more of what can be done, I would feedback from anyone on this.

File: app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java [hunk 1]
```diff
@@ -976,33 +992,48 @@ private void sendFetchTimelineRequest(@Nullable String maxId, @Nullable String s
...
-            Callback<List<Status>> callback = new Callback<List<Status>>() {
-                @Override
-                public void onResponse(@NonNull Call<List<Status>> call, @NonNull Response<List<Status>> response) {
-                    if (response.isSuccessful()) {
-                        onFetchTimelineSuccess(liftStatusList(response.body()), fetchEnd, pos);
-                    } else {
-                        onFetchTimelineFailure(new Exception(response.message()), fetchEnd, pos);
-                    }
-                }
+            Single<?> result = Single
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java]: Maybe I shouldn't do this and should instead convert all of the timeline requests into `Single`s.

**Comment by charlag** [comment]
Updated it a little bit because I've found some ridiculous bugs.

**Pull request closed without merging**